server/services/store/sqlstore/migrate.go

```diff
@@ -213,6 +225,194 @@ func (s *SQLStore) Migrate() error {
+	// for SQLite. Hence, the separate function
+
+	query := fmt.Sprintf("PRAGMA table_info(\"%sschema_migrations\");", s.tablePrefix)
+	rows, err := s.db.Query(query)
+	if err != nil {
+		s.logger.Error("SQLite - failed to check for columns in schema_migrations table", mlog.Err(err))
+		return false, err
+	}
+
+	defer s.CloseRows(rows)
+
+	data := [][]*string{}
+	for rows.Next() {
+		// PRAGMA returns 6 columns
+		row := make([]*string, 6)
+
+		err := rows.Scan(
+			&row[0],
+			&row[1],
+			&row[2],
+			&row[3],
+			&row[4],
+			&row[5],
+		)
+		if err != nil {
+			s.logger.Error("error scanning rows from SQLite schema_migrations table definition", mlog.Err(err))
+			return false, err
+		}
+
+		data = append(data, row)
+	}
+
+	nameColumnFound := false
+	for _, row := range data {
+		if len(row) >= 2 && *row[1] == "dirty" {
+			nameColumnFound = true
+			break
+		}
+	}
+
+	return nameColumnFound, nil
+}
+
+func (s *SQLStore) getLegacySchemaVersion() (uint32, error) {
+	query := s.getQueryBuilder(s.db).
+		Select("version").
+		From(s.tablePrefix + "schema_migrations")
+
+	row := query.QueryRow()
+
+	var version uint32
+	if err := row.Scan(&version); err != nil {
+		s.logger.Error("error fetching legacy schema version", mlog.Err(err))
+		s.logger.Error("getLegacySchemaVersion err " + err.Error())
+		return version, err
+	}
+
+	return version, nil
+}
+
+func (s *SQLStore) createTempSchemaTable() error {
+	// squirrel doesn't support DDL query in query builder
+	// so, we need to use a plain old string
+	query := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (Version bigint NOT NULL, Name varchar(64) NOT NULL, PRIMARY KEY (Version))", s.tablePrefix+tempSchemaMigrationTableName)
+	if _, err := s.db.Exec(query); err != nil {
+		s.logger.Error("failed to create temporary schema migration table", mlog.Err(err))
+		s.logger.Error("createTempSchemaTable error  " + err.Error())
+		return err
+	}
+
+	return nil
+}
+func (s *SQLStore) populateTempSchemaTable(migrations []*models.Migration, legacySchemaVersion uint32) error {
+	query := s.getQueryBuilder(s.db).
+		Insert(s.tablePrefix+tempSchemaMigrationTableName).
+		Columns("Version", "Name")
+
+	for _, migration := range migrations {
+		// migrations param contains both up and down variant for
+		// each migration. Skipping for either one (down in this case)
+		// to process a migration only a single time.
+		if migration.Direction == models.Down {
+			continue
+		}
+
+		if migration.Version > legacySchemaVersion {
+			break
+		}
+
+		query = query.Values(migration.Version, migration.Name)
+	}
+
+	if _, err := query.Exec(); err != nil {
+		s.logger.Error("failed to insert migration records into temporary schema table", mlog.Err(err))
+		return err
+	}
+
+	return nil
+}
+
+func (s *SQLStore) useNewSchemaTable() error {
+	// first delete the old table, then
+	// rename the new table to old table's name
+
+	// deleting old table
+	query := "DROP TABLE " + s.tablePrefix + "schema_migrations"
```

Comment:
Before deleting, should we just try to rename this? Then use new one, then delete this one.

Reply:
Done.